### PR TITLE
[libavtp] Add builder

### PR DESCRIPTION
**Pull request by bramtayl**

### CODE REVIEW

**Comment by bramtayl** [comment]
Well, this works on a handful of platforms at least? It's for one of the alsa-plugins but I'm not sure how useful it is

**Comment by bramtayl** [comment]
I'm pretty sure only glibc and Linux are supported

File: L/libavtp/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd libavtp-*
+mkdir build
+cd build
+meson --cross-file=${MESON_TARGET_TOOLCHAIN}
+ninja
```
Comment (giordano) [inline L/libavtp/build_tarballs.jl:20]:
```suggestion
cd $WORKSPACE/srcdir/libavtp-*
mkdir build
cd build
meson --cross-file=${MESON_TARGET_TOOLCHAIN}
ninja -j${nproc}
```

File: L/libavtp/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+platforms = filter!(platform -> Sys.islinux(platform) && libc(platform) == "glibc", supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libavtp", :libavtp)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="cmocka_jll", uuid="f83fd561-6387-5ecc-9835-b38c8eaffb11"))
```
Comment (giordano) [inline L/libavtp/build_tarballs.jl:35]: What is this library used for?  libavtp doesn't seem to link to it
```console
% ldd x86_64-linux-gnu/lib/libavtp.so
        linux-vdso.so.1 (0x00007ffe3b3c3000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f5ef5444000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f5ef5844000)
```

Comment (bramtayl) [reply L/libavtp/build_tarballs.jl:35]: No idea, but meson sends the message

> Run-time dependency cmocka found: YES 1.1.5

?

Comment (giordano) [reply L/libavtp/build_tarballs.jl:34]: Seems to be used only for testing: https://github.com/Avnu/libavtp/blob/43a1f61565313166254ab92a39b53a005ac99ee2/HACKING.md#running-unit-tests

Comment (giordano) [reply L/libavtp/build_tarballs.jl:34]: It isn't a runtime dependency in archlinux either: https://archlinux.org/packages/extra/x86_64/libavtp/

File: L/libavtp/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+sources = [
+    ArchiveSource("https://github.com/Avnu/libavtp/archive/refs/tags/v$(version).tar.gz", "9c1587431281a8d2404060e4bfd39f5cf0f211be6251e6afb1100e6148c1e591")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/libavtp-*
+mkdir build
+cd build
+meson --cross-file=${MESON_TARGET_TOOLCHAIN}
```
Comment (giordano) [inline L/libavtp/build_tarballs.jl:18]: https://github.com/Avnu/libavtp/blob/43a1f61565313166254ab92a39b53a005ac99ee2/meson.build#L45-L49
```suggestion
meson --cross-file=${MESON_TARGET_TOOLCHAIN} -Dtests=disabled
```

File: L/libavtp/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+platforms = filter!(platform -> Sys.islinux(platform) && libc(platform) == "glibc", supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libavtp", :libavtp)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="cmocka_jll", uuid="f83fd561-6387-5ecc-9835-b38c8eaffb11"))
```
Comment (giordano) [inline L/libavtp/build_tarballs.jl:34]:
```suggestion
dependencies = Dependency[
```

File: L/libavtp/build_tarballs.jl (hunk 3 above)
Comment (bramtayl) [reply L/libavtp/build_tarballs.jl]: WARNING: Unknown options: "tests"?

Comment (giordano) [reply L/libavtp/build_tarballs.jl]: https://github.com/Avnu/libavtp/blob/43a1f61565313166254ab92a39b53a005ac99ee2/meson_options.txt#L3 and the link above :shrug:

Comment (bramtayl) [reply L/libavtp/build_tarballs.jl]: I think the option was added after the most recent release, https://github.com/Avnu/libavtp/pull/24

Comment (giordano) [reply L/libavtp/build_tarballs.jl]: Then make cmocka a build dependency....

File: L/libavtp/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# platforms are passed in on the command line
+platforms = filter!(platform -> Sys.islinux(platform) && libc(platform) == "glibc", supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libavtp", :libavtp)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
```
Comment (giordano) [inline L/libavtp/build_tarballs.jl:33]:
```suggestion
dependencies = [
```

**Pull request merged**